src/tree-internal.component.ts

```diff
@@ -162,28 +182,45 @@ export class TreeInternalComponent implements OnInit, OnChanges, OnDestroy, Afte
   }
 
   public ngOnDestroy(): void {
-    if (get(this.tree, 'node.id', '')) {
+    if (get(this.tree, 'node.id', '') && !(this.tree.parent && this.tree.parent.children.indexOf(this.tree) > -1)) {
       this.treeService.deleteController(this.tree.node.id);
     }
 
     this.subscriptions.forEach(sub => sub && sub.unsubscribe());
   }
 
-  private swapWithSibling(sibling: Tree, tree: Tree): void {
-    tree.swapWithSibling(sibling);
-    this.treeService.fireNodeMoved(sibling, sibling.parent);
+  private moveSibling(sibling: Tree, tree: Tree, position: DropPosition): void {
+    const previousPositionInParent = sibling.positionInParent;
+    if (position === DropPosition.Above) {
+      tree.moveSiblingAbove(sibling);
+    } else {
+      tree.moveSiblingBelow(sibling);
+    }
+    this.treeService.fireNodeMoved(sibling, sibling.parent, previousPositionInParent);
   }
 
-  private moveNodeToThisTreeAndRemoveFromPreviousOne(e: NodeDraggableEvent, tree: Tree): void {
-    this.treeService.fireNodeRemoved(e.captured.tree);
```

Comment:
I'm a bit concerned about the implications of this change upstream(valor-software/ng2-tree).

Reply:
I agree, but this was causing issues in Studio where the `NodeRemovedEvent` was being picked up and a remove confirmation dialog was shown. There was no way to determine whether a node was being removed from one tree and re-added to another tree (i.e. being moved), or it was actually being removed permanently. I don't think there is a non-breaking solution here, but I am open to suggestions for a different solution.

There're probably a few more breaking changes like this one, some of which we could make configurable, but I think there will be a few unavoidable ones. This component has become quite specific to Arria's needs, so I am wondering how likely it is that it will ever be pulled into the main repo. I am thinking whether a separate tech-debt ticket should be created to deal with pushing our changes back into the main repo (and dealing with any issues related to that).

Reply:
I agree with you in that it's gotten more and more specific to our needs. Given the timeframe, we don't have enough time to tackle this right now, so guess we should put it off for later.